src/lib/systemlib/system_params.c

```diff
@@ -77,6 +77,19 @@ PARAM_DEFINE_INT32(SYS_AUTOCONFIG, 0);
  */
 PARAM_DEFINE_INT32(SYS_HITL, 0);
 
+/**
+ * Enable SIH mode on next boot
```

Comment:
@bkueng Should this be with all the other SIH parameters? I do appreciate this is enabling new system behaviour, but I would have thought it better to group with the associated module?

Reply:
I guess I prefer extending SYS_HITL=2 instead of adding another parameter for this. Given that we need to disable sensors in both cases.

Reply:
I am not against setting SYS_HITL=2. 

[rcS](https://github.com/PX4/Firmware/pull/11400/files#diff-e5c16c5209413f30e964883f74027b51L327) will need to be modified as
```suggestion
	if param greater SYS_HITL 0
	then
		set OUTPUT_MODE hil
		sensors start -h
		commander start --hil
		# disable GPS
		param set GPS_1_CONFIG 0

		# start the simulator in hardware if needed
		if  param compare SYS_HITL 2
		then
			sih start
		fi
```
and [rc.interface](https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/rc.interface#L205) will need to be modified as
```suggestion
if param greater SYS_HITL 0
```
of course, the comments on [system_params.c](https://github.com/PX4/Firmware/pull/11400/files/8206d6c73abe28ae4720d25902efbd2cd66ec98e#diff-58f0a6ba05be221eaead7a8dd240c391L68) will need to be updated 
```suggestion
/**
 * Enable HITL mode on next boot
 *
 * While enabled the system will boot in HITL mode and not enable all sensors and checks.
 * When disabled the same vehicle can be flown normally.
 * Setting SYS_HITL to 1 will enable the hardware in the loop.
 * Setting SYS_HITL to 2 will enable the simulation in hardware (SIH).
 *
 * @boolean
 * @reboot_required true
 *
 * @group System
 */
PARAM_DEFINE_INT32(SYS_HITL, 0);
```

Any other file using SYS_HITL, that will need to be modified?

Reply:
> Any other file using SYS_HITL, that will need to be modified?

That looks correct, except for the param defintion. It should be something like:
```
/**
 * Enable HITL/SIH mode on next boot
 *
 * While enabled the system will boot in Hardware-In-The-Loop (HITL) or Simulation-In-Hardware (SIH) mode and not enable all sensors and checks.
 * When disabled the same vehicle can be flown normally.
 *
 * @value 0 HITL disabled
 * @value 1 HITL enabled
 * @value 2 SIH enabled
 * @reboot_required true
 *
```